source/reference/ejson/deserialize.txt

```diff
@@ -0,0 +1,200 @@
+.. _mongosh-ejson-deserialize:
+
+===================
+EJSON.deserialize()
+===================
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. facet::
+   :name: programming_language
+   :values: shell
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. include:: /includes/links-urls/ejson.rst
+
+The ``EJSON.deserialize()`` method converts Extended JSON objects to
+JavaScript objects.
+
+Syntax
+------
+
+The method has this syntax:
+
+.. code-block:: javascript
+   :copyable: false
+
+   EJSON.deserialize( object, [ options ] )
+
+Method Fields
+-------------
+
+The method takes the following fields:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Field
+     - Type
+     - Necessity
+     - Description
+
+   * - ``object``
+     -  EJSON object
+     -  Required
+     -  EJSON object to convert. For example, an array of documents.
+
+   * - ``options``
+     -  string
+     -  Optional
+     -  Modifies the output object :ref:`types <mongo-shell-data-type>`.
+        The only option is ``{ relaxed: <boolean> }``.
+ 
+        .. list-table::
+           :header-rows: 1
+           :widths: 30 70
+
+           * - Boolean Value
+             - Description
+
+           * - ``true``
+             - Return JSON object types. Default is ``true``.
+
+           * - ``false``
+             - Return BSON object types.
+
+Behavior
+--------
+
+You can run ``EJSON.deserialize()`` from an interactive ``mongosh``
+session or from the system command line using ``--eval``.
+
+To run ``EJSON.deserialize()`` from an interactive ``mongosh`` session,
+use:
+
+.. code-block:: javascript
+   :copyable: false
+
+   EJSON.deserialize( object, [ options ] )
+
+To run ``EJSON.deserialize()`` from the system command line, use:
+
+.. code-block:: shell
+   :copyable: false
+
+   mongosh --eval "EJSON.deserialize( object, [ options ] )"
+
+Examples
+--------
+
+Create the ``sales`` collection for the examples:
+
+.. code-block:: javascript
+
+   db.sales.insertMany( [ 
+      { custId: 345, purchaseDate: ISODate("2023-07-04"),
+        quantity: 4, cost: Decimal128("100.60") },
+      { custId: 346, purchaseDate: ISODate("2023-07-12"),
+        quantity: 3, cost: Decimal128("175.45") },
+      { custId: 486, purchaseDate: ISODate("2023-08-01"),
+        quantity: 9, cost: Decimal128("200.53") }
+   ] )
+
+Interactive Mongo Shell EJSON.deserialize() Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example retrieves the ``sales`` documents as an array and
+stores the results in the ``salesCollection`` object:
+
+.. code-block:: javascript
+
+   salesCollection = EJSON.deserialize( db.sales.find().toArray() )
```

Comment:
Similar to `EJSON.parse()`, the use cases revolve around consuming external data, processing it, and then being able to interact with them like database contents.

Calling `EJSON.deserialize()` on objects coming from the database doesn’t make sense, the database already returns BSON objects, so there’s no need to convert them “back” to BSON objects.

The example below (reading data from disk and inserting it into the database) is definitely a better one.